internal/serverinstall/k8s.go

```diff
@@ -83,176 +79,145 @@ func (i *K8sInstaller) Install(
+	client.GenerateName = false
+	client.NameTemplate = ""
+	client.OutputDir = ""
+	client.Atomic = false
+	client.SkipCRDs = false
+	client.SubNotes = true
+	client.DisableOpenAPIValidation = false
+	client.Replace = false
+	client.Description = ""
+	client.CreateNamespace = true
+	s.Update("Helm install created")
+	s.Status(terminal.StatusOK)
+	s.Done()
 
-		secretsClient := clientset.CoreV1().Secrets(i.config.namespace)
-		_, err = secretsClient.Create(context.TODO(), &secretData, metav1.CreateOptions{})
+	var version string
+	if i.config.version == "" {
+		tags, err := helminstallutil.GetLatestHelmChartVersion(ctx)
 		if err != nil {
-			ui.Output(
-				"Error creating Kubernetes secret from file: %s", clierrors.Humanize(err),
-				terminal.WithErrorStyle(),
-			)
+			opts.UI.Output("Error getting latest tag of Waypoint helm chart.", terminal.WithErrorStyle())
 			return nil, err
 		}
-
-		s.Done()
-		s = sg.Add("")
-	}
-
-	// Do some probing to see if this is OpenShift. If so, we'll switch the config for the user.
-	// Setting the OpenShift flag will short circuit this.
-	if !i.config.openshift {
-		s.Update("Gathering information about the Kubernetes cluster...")
-		namespaceClient := clientset.CoreV1().Namespaces()
-		_, err := namespaceClient.Get(context.TODO(), "openshift", metav1.GetOptions{})
-		isOpenShift := err == nil
-
-		// Default namespace in OpenShift acts like a regular K8s namespace, so we don't want
-		// to remove fsGroup in this case.
-		if isOpenShift && i.config.namespace != "default" {
-			s.Update("OpenShift detected. Switching configuration...")
-			i.config.openshift = true
-		}
+		version = *tags[0].Name
+	} else {
+		version = i.config.version
 	}
 
-	// Decode our configuration
-	statefulset, err := newStatefulSet(i.config, opts.ServerRunFlags)
+	s = sg.Add("Locating chart...")
+	path, err := client.LocateChart("https://github.com/hashicorp/waypoint-helm/archive/refs/tags/"+version+".tar.gz", settings)
 	if err != nil {
-		ui.Output(
-			"Error generating statefulset configuration: %s", clierrors.Humanize(err),
-			terminal.WithErrorStyle(),
-		)
 		return nil, err
 	}
+	s.Update("Helm chart located")
+	s.Status(terminal.StatusOK)
+	s.Done()
 
-	service, err := newService(i.config)
+	s = sg.Add("Loading Helm chart...")
+	c, err := loader.Load(path)
 	if err != nil {
-		ui.Output(
-			"Error generating service configuration: %s", clierrors.Humanize(err),
-			terminal.WithErrorStyle(),
-		)
 		return nil, err
 	}
+	s.Update("Helm chart loaded")
+	s.Status(terminal.StatusOK)
+	s.Done()
 
-	if i.config.odrServiceAccountInit {
-		s.Done()
-		err := i.initServiceAccount(ctx, clientset, sg)
-		if err != nil {
-			return nil, err
-		}
-		s = sg.Add("")
-	}
-
-	s.Update("Creating Kubernetes resources...")
-
-	serviceClient := clientset.CoreV1().Services(i.config.namespace)
-	_, err = serviceClient.Create(context.TODO(), service, metav1.CreateOptions{})
+	imageRef, err := dockerparser.Parse(i.config.serverImage)
```

Comment:
Just a guess, but I think this is trimming off the registry address for an image. We need to preserve this otherwise you won't be able to pull from custom container registries